app/controllers/settings_controller.py

```diff
@@ -570,6 +571,10 @@ def _update_model_from_view(self) -> None:
             self.settings.sorting_algorithm = "Alphabetical"
         elif self.settings_dialog.sorting_topological_radio.isChecked():
             self.settings.sorting_algorithm = "Topological"
+        if self.settings_dialog.db_use_dependency_for_sorting_checkbox.isChecked():
+            self.settings.dependency_for_sorting = "True"
+        else:
+            self.settings.dependency_for_sorting = "False"
```

Comment:
I'd just do:

```suggestion
self.settings.dependency_for_sorting = self.settings_dialog.db_use_dependency_for_sorting_checkbox.isChecked()
```
instead since you're storing the setting as a bool